ecc/bls12381/pair.go

```diff
@@ -0,0 +1,96 @@
+package bls12381
+
+import "github.com/cloudflare/circl/ecc/bls12381/ff"
+
+// Pair calculates the ate-pairing of P and Q.
+func Pair(P *G1, Q *G2) *Gt { return finalExp(miller(P, Q)) }
```

Comment:
What I really want is products of pairings with exponents in them.

Reply:
This 

<img src="https://render.githubusercontent.com/render/math?math=\product_i%20e(P_i,Q_i)^{n_i}">

or

<img src="https://render.githubusercontent.com/render/math?math=\left(\product_i%20e(P_i,Q_i)\right)^{n}">

Reply:
The former. We save time by integrating all those final exponents and the local ones, but doing it in constant time might be tricky. I'll see if I can extract something more specialized that would help.

Reply:
Already included.